system/jlib/jthread.cpp

```diff
@@ -2565,3 +2616,66 @@ unsigned threadLogID()  // for use in logging
 #endif
     return (unsigned)(memsize_t) GetCurrentThreadId(); // truncated in 64bit
 }
+
+void PerfTracer::start()
+{
+#ifdef __linux__
+    dostart(1000000);
+#else
+    UNIMPLEMENTED;
+#endif
+}
+
+void PerfTracer::dostart(unsigned seconds)
+{
+#ifdef __linux__
+    pipe.setown(createPipeProcess());
+    pipe->setAllowTrace();
+    VStringBuffer cmd("doperf %u %u %f", GetCurrentProcessId(), seconds, interval);
+    if (!pipe->run(nullptr, cmd, ".", false, true, false, 1024*1024))
+    {
+        pipe.clear();
+        throw makeStringException(0, "Failed to run doperf");
+    }
+#else
+    UNIMPLEMENTED;
+#endif
+}
+
+void PerfTracer::stop()
+{
+#ifdef __linux__
+    assertex(pipe);
+    ::kill(pipe->getProcessHandle(), SIGINT);
+    dostop();
+#else
+    UNIMPLEMENTED;
+#endif
+}
+
+void PerfTracer::traceFor(unsigned seconds)
+{
+#ifdef __linux__
+    dostart(seconds);
+    dostop();
+#else
+    UNIMPLEMENTED;
+#endif
+}
+
+void PerfTracer::dostop()
+{
+#ifdef __linux__
+    char buf[1024];
```

Comment:
Should this use > 1024?

Reply:
It uses the same as the other places we read output from pipe. I don't think it really matters what it uses (within reason).